### PR TITLE
Sign vrt-strings

### PR DESCRIPTION
**Pull request by TomAugspurger**
xref #32, which also includes feature requests for signing files. This handles the in-memory string portion of things.

The imagined workflow here is something like

```console
$ gdal_translate "STACIT:\"https://planetarycomputer.microsoft.com/api/stac/v1/search?collections=naip&bbox=-100,40,-99,41&datetime=2019-01-01T00:00:00Z%2F..\":asset=image" out.vrt
```

And then using it from Python like

```python
In [1]: import pathlib, planetary_computer, rioxarray

In [2]: signed_vrt = planetary_computer.sign(pathlib.Path("out.vrt").read_text())

In [3]: ds = rioxarray.open_rasterio(signed_vrt)

In [4]: ds
Out[4]:
<xarray.DataArray (band: 4, y: 25023, x: 161196)>
[16134430032 values with dtype=uint8]
Coordinates:
  * band         (band) int64 1 2 3 4
  * x            (x) float64 4.09e+05 4.09e+05 4.09e+05 ... 5.057e+05 5.057e+05
  * y            (y) float64 4.436e+06 4.436e+06 ... 4.421e+06 4.421e+06
    spatial_ref  int64 0
Attributes:
    scale_factor:  1.0
    add_offset:    0.0
    long_name:     ('Red', 'Green', 'Blue', 'NIR')
```

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Seems like the pinned version of black didn't work with the latest Python 3.9:

```
Running black...
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.8/x64/bin/black", line 5, in <module>
    from black import patched_main
  File "/opt/hostedtoolcache/Python/3.9.8/x64/lib/python3.9/site-packages/black/__init__.py", line 52, in <module>
    from typed_ast import ast3, ast27
  File "/opt/hostedtoolcache/Python/3.9.8/x64/lib/python3.9/site-packages/typed_ast/ast3.py", line 40, in <module>
    from typed_ast import _ast3
ImportError: /opt/hostedtoolcache/Python/3.9.8/x64/lib/python3.9/site-packages/typed_ast/_ast3.cpython-39-x86_64-linux-gnu.so: undefined symbol: _PyUnicode_DecodeUnicodeEscape
```

I bumped those deps in https://github.com/microsoft/planetary-computer-sdk-for-python/pull/33/commits/c937cb112fd5a3b8e263cba7897fc5880e73a0ae

File: planetary_computer/sas.py [hunk 1]
```diff
@@ -93,6 +107,14 @@ def sign_url(url: str) -> str:
     return token.sign(url).href
 
 
+def _repl_vrt(m: re.Match) -> str:
+    return sign_url(m.string[slice(*m.span())])
+
+
+def sign_vrt_string(vrt: str) -> str:
```
Comment (TomAugspurger) [inline planetary_computer/sas.py:114]: Needs a docstring.

File: planetary_computer/sas.py [hunk 2]
```diff
@@ -84,6 +91,13 @@ def sign_url(url: str) -> str:
     Returns:
         str: The signed HREF
     """
+    if is_vrt_string(url):
+        return sign_vrt_string(url)
+    else:
+        return sign_url(url)
+
+
+def sign_url(url: str) -> str:
```
Comment (TomAugspurger) [inline planetary_computer/sas.py:100]: Needs a docstring.

File: planetary_computer/sas.py [hunk 3]
```diff
@@ -73,7 +80,7 @@ def sign(obj: Any) -> Any:
 
 
 @sign.register(str)
-def sign_url(url: str) -> str:
+def sign_string(url: str) -> str:
```
Comment (TomAugspurger) [inline planetary_computer/sas.py:83]: This docstring should be updated that it's for URLs or VRT-like strings.

File: planetary_computer/sas.py (hunk 2 above)
Comment (TomAugspurger) [reply planetary_computer/sas.py:100]: And I wonder if the implementation can / should use the regex-based implementation required by VRTs. I'll need to time it.

File: planetary_computer/sas.py [hunk 4]
```diff
@@ -73,13 +80,41 @@ def sign(obj: Any) -> Any:
...
+
+            GDAL VRTs can combine many data sources into a single mosaic. A VRT can be
+            built quickly from the GDAL STACIT driver
+            https://gdal.org/drivers/raster/stacit.html. Each URL to Azure Blob Storage
+            within the VRT is signed.
+
+    Returns:
+        str: The signed HREF or VRT
+    """
+    if is_vrt_string(url):
```
Comment (TomAugspurger) [inline planetary_computer/sas.py:102]: For the record, this doesn't seem to have a measurable effect on the timing of `planetary_computer.sign(url)`:

```python
# main, after signing a URL to get a cached token
In [5]: %timeit planetary_computer.sign(url)
12.2 µs ± 1.97 µs per loop (mean ± std. dev. of 7 runs, 100000 loops each)

# HEAD
In [4]: %timeit planetary_computer.sign(url)
11.3 µs ± 1.7 µs per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

**Comment by TomAugspurger** [comment]
FYI @lossyrob, I plan to merge this sometime today. It shouldn't have any effect on existing behavior and I think is pretty uncontroversial.

**Review by lossyrob — APPROVED** [review]
🚀

**Pull request merged**